### DISCUSSION TITLE
Abstracciones en Clean Architecture

### DISCUSSION THREAD

**Discussion by David-Lazaro-Fernandez** [gittogethers/octogatosconf22#78] [Talks in Spanish]
***Ponente: Victor A Higuita C***
> Cómo construir una solución de software en clean architecture, entenderlo y no morir en el intento.

**Comment by TsukiZombina** [comment]
¿dónde encuentro los retos?

**Reply by harvic3** [reply]
Hola @TsukiZombina 
Los retos en las ultimas página de la presentación y la puedes obtener en el enlace:
https://github.com/harvic3/octogatos-conf-2022/blob/main/Abstracciones%20en%20CA%20(Octogatos%20Conf%2022).pdf

**Comment by davidcoachdev** [comment]
super explicación podrías subir el material por este medio para poder revisar la más a fondo y de nuevo super explicación

**Reply by harvic3** [reply]
Qué bueno que te haya gustado, y con gusto. 
El repo de la presentación es:
https://github.com/harvic3/octogatos-conf-2022
El repo para los retos es:
https://github.com/harvic3/nodetskeleton/tree/feature/queue-bus-events

**Reply by davidcoachdev** [reply]
Gracias.

**Comment by dahidalgo** [comment]
está en algún lenguaje o cual es el más recomendado para hacer los retos?

**Reply by harvic3** [reply]
El proyecto está hecho en **TypeScript** usando node.js

El repo de la presentación es:
https://github.com/harvic3/octogatos-conf-2022
El repo para los retos es:
https://github.com/harvic3/nodetskeleton/tree/feature/queue-bus-events

**Comment by harvic3** [comment]
Pregunta/Respuesta del chat de twitch:
**Vinoconpritty**: Arquitecto de Software es algo que se puede estudiar, o viene más por el lado de la experiencia haciendo? Asumo que todo suma, pero la capacidad de Abstracción por ejemplo viene con los años :(?

Rta: si es posible llegar a ser arquitecto de software sin haber tenido mucha experiencia con el código, sin embargo eso sólo sería posible si tienes una alta capacidad (mente brillante) para manejar las abstracciones de forma natural, modelar sistemas a nivel macro y micro, crear flujos de comunicación entre diferentes servicios, algo que se consigue con la practica y la experiencia. Sin  embargo dado el amplio abanico de conceptos en cuanto a software se refiere, tendrías que leer mucho, básicamente tragar literatura por montones, lo que consigues de forma natural a través de la experiencia y de esta forma creas una carrera de crecimiento profesional de acuerdo a tus gustos e intereses.

**Comment by harvic3** [comment]
Les dejo los recursos para que monten REDIS si aún no lo han hecho 

Requisitos:
1. Docker

Crear instancia de Redis
> docker run --name **container-name** -p 6379:6379 -d redis
Or
> docker run --rm --name **image-name** --net **net-name** -p 6379:6379 -d redis
